library/src/dotty/source/Position.scala

```diff
@@ -0,0 +1,23 @@
+package dotty.source
+
+object Position {
+
+  /** Line number */
+  final class LineNumber(val value: Int) extends AnyVal {
+    override def toString: String = value.toString
+  }
+
+  /** Path of a source file */
+  final class SourcePath(val value: String) extends AnyVal {
+    override def toString: String = value
+  }
+
+  /** LineNumber containing the line number of its call site.
+   *  The line number is 1-indexed.
+   */
+  implicit def thisLine: LineNumber = new LineNumber(0)
```

Comment:
I wonder if this should be inside `object LineNumber` to allow non-wildcard imports

Reply:
Good point, I will change this.